hawc/apps/epiv2/views.py

```diff
@@ -11,22 +17,226 @@ class DesignCreate(BaseCreate):
     model = models.Design
     form_class = forms.DesignForm
 
+    def get_success_url(self):
+        return reverse("epiv2:design_update", args=[self.object.id])
+
 
 class DesignUpdate(BaseUpdate):
     success_message = "Study-population updated."
     parent_model = Study
     parent_template_name = "study"
     model = models.Design
     form_class = forms.DesignForm
+    template_name = "epiv2/design_update.html"
+
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .prefetch_related(
+                "exposures",
+                "outcomes",
+                "chemicals",
+                "criteria",
+                "adjustment_factors",
+                Prefetch(
+                    "exposure_levels",
+                    queryset=models.ExposureLevel.objects.select_related(
+                        "chemical", "exposure_measurement"
+                    ),
+                ),
+                Prefetch(
+                    "data_extractions",
+                    queryset=models.DataExtraction.objects.select_related(
+                        "adjustment_factor", "outcome", "exposure_level"
+                    ),
+                ),
+            )
+        )
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context["exposures"] = self.object.exposures.all()
+        context["outcomes"] = self.object.outcomes.all()
+        context["chemical"] = self.object.chemicals.all()
+        context["criteria"] = self.object.criteria.all()
+        context["exposure_levels"] = self.object.exposure_levels.all()
+        context["adjustment_factors"] = self.object.adjustment_factors.all()
+        context["data_extractions"] = self.object.data_extractions.all()
+        return context
 
 
 class DesignDetail(BaseDetail):
     model = models.Design
 
+    def get_queryset(self):
+        return (
+            super()
+            .get_queryset()
+            .prefetch_related(
+                "exposures",
+                "outcomes",
+                "chemicals",
+                "criteria",
+                "adjustment_factors",
+                Prefetch(
+                    "exposure_levels",
+                    queryset=models.ExposureLevel.objects.select_related(
+                        "chemical", "exposure_measurement"
+                    ),
+                ),
+                Prefetch(
+                    "data_extractions",
+                    queryset=models.DataExtraction.objects.select_related(
+                        "adjustment_factor", "outcome", "exposure_level"
+                    ),
+                ),
+            )
+        )
+
+    def get_context_data(self, **kwargs):
```

Comment:
I wonder if it's better to do it this way, or to update templates to just use the `{{object.exposures.all}}` etc, what do you think?

Reply:
I decided to update the templates, the views look a lot cleaner that way.